x-pack/plugins/ml/public/data_frame/components/define_pivot/define_pivot_form.tsx

```diff
@@ -66,6 +66,99 @@ export function getDefaultPivotState(kibanaContext: KibanaContextValue): DefineP
     valid: false,
   };
 }
+export function isAggNameConflict(
+  aggName: AggName,
+  aggList: PivotAggsConfigDict,
+  groupByList: PivotGroupByConfigDict
+) {
+  if (aggList[aggName] !== undefined) {
+    toastNotifications.addDanger(
+      i18n.translate('xpack.ml.dataframe.definePivot.aggExistsErrorMessage', {
+        defaultMessage: `An aggregation configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      })
+    );
+    return true;
+  }
+
+  if (groupByList[aggName] !== undefined) {
+    toastNotifications.addDanger(
+      i18n.translate('xpack.ml.dataframe.definePivot.groupByExistsErrorMessage', {
+        defaultMessage: `A group by configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      })
+    );
+    return true;
+  }
+
+  let conflict = false;
+
+  // check the new aggName against existing aggs and groupbys
+  const aggNameSplit = aggName.split('.');
+  let aggNameCheck: string;
+  aggNameSplit.forEach(aggNamePart => {
+    aggNameCheck = aggNameCheck === undefined ? aggNamePart : `${aggNameCheck}.${aggNamePart}`;
```

Comment:
Nit - might be good to wrap this in parens for readability
`aggNameCheck = (aggNameCheck === undefined ? aggNamePart : `${aggNameCheck}.${aggNamePart}`);`

Reply:
I'm afraid `prettier` enforces that :)